alf/algorithms/agent.py

```diff
@@ -141,20 +141,28 @@ def predict(self, time_step: ActionTimeStep, state: AgentState):
 
         return PolicyStep(action=rl_step.action, state=new_state, info=())
 
-    def rollout(self,
-                time_step: ActionTimeStep,
-                state: AgentState,
-                with_experience=False):
+    def greedy_predict(self, time_step: ActionTimeStep, state=None, eps=0.1):
+        observation = self._encode(time_step)
+
+        new_state = AgentState()
+
+        rl_step = self._rl_algorithm.greedy_predict(
+            time_step._replace(observation=observation), state.rl)
+        new_state = new_state._replace(rl=rl_step.state)
+
+        return PolicyStep(action=rl_step.action, state=new_state, info=())
+
+    def rollout(self, time_step: ActionTimeStep, state: AgentState):
         """Rollout for one step."""
         new_state = AgentState()
         info = AgentInfo()
         observation = self._encode(time_step)
 
+        # TODO
+        # avoid computing this when rl_algorithm do not need preprocess experience
```

Comment:
What do you mean by this comment? I believe that for any algorithm, icm rewards should always be computed during rollout.

Reply:
Removed,  I thought it should be  calculated  at  `train_step` for some off policy algorithms that do not need preprocessing experiences